diff -up augeas-0.9.0/lenses/fstab.aug.orig augeas-0.9.0/lenses/fstab.aug
--- augeas-0.9.0/lenses/fstab.aug.orig	2011-07-25 22:34:14.000000000 +0100
+++ augeas-0.9.0/lenses/fstab.aug	2012-05-11 17:05:27.499437098 +0100
@@ -16,7 +16,7 @@ module Fstab =
   let spec    = /[^,# \n\t][^ \n\t]*/
 
   let comma_sep_list (l:string) =
-    let value = [ label "value" . Util.del_str "=" . store Rx.neg1 ] in
+    let value = [ label "value" . Util.del_str "=" . ( store Rx.neg1 )? ] in
       let lns = [ label l . store optlabel . value? ] in
          Build.opt_list lns comma
 
diff -up augeas-0.9.0/lenses/tests/test_fstab.aug.orig augeas-0.9.0/lenses/tests/test_fstab.aug
--- augeas-0.9.0/lenses/tests/test_fstab.aug.orig	2011-07-25 22:34:14.000000000 +0100
+++ augeas-0.9.0/lenses/tests/test_fstab.aug	2012-05-11 17:05:27.499437098 +0100
@@ -102,6 +102,20 @@ module Test_fstab =
     { "passno" = "0" } }
 
 
+  (* Parse when empty option value given, only equals sign *)
+  test Fstab.lns get "//host.example.org/a_share /mnt cifs defaults,ro,password= 0 0\n" =
+  { "1"
+    { "spec" = "//host.example.org/a_share" }
+    { "file" = "/mnt" }
+    { "vfstype" = "cifs" }
+    { "opt" = "defaults" }
+    { "opt" = "ro" }
+    { "opt" = "password"
+      { "value" }
+    }
+    { "dump" = "0" }
+    { "passno" = "0" }
+  }
 (* Local Variables: *)
 (* mode: caml       *)
 (* End:             *)
