commit cf9338b7111b8cfbcb6c6a5eea776b86106ee3d6
Author: Raphaël Pinson <raphink@gmail.com>
Date:   Thu Dec 8 17:59:55 2011 +0100

    Debian-based distributions use /etc/mdadm/mdadm.conf

diff --git a/lenses/mdadm_conf.aug b/lenses/mdadm_conf.aug
index 2f20158..635a332 100644
--- a/lenses/mdadm_conf.aug
+++ b/lenses/mdadm_conf.aug
@@ -274,5 +274,6 @@ let lns = (comment
            | part_policy )*
 
 let filter     = incl "/etc/mdadm.conf"
+               . incl "/etc/mdadm/mdadm.conf"
 
 let xfm        = transform lns filter
