From 5f109ae5922fc8b4f115faaa2f14eb836cec1552 Mon Sep 17 00:00:00 2001
From: David Lutterkort <lutter@redhat.com>
Date: Mon, 8 Aug 2011 16:02:26 -0700
Subject: [PATCH 3/3] * examples/fadot.c: accept -h option to print usage

---
 examples/fadot.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/examples/fadot.c b/examples/fadot.c
index 0edd813..cb514cc 100644
--- a/examples/fadot.c
+++ b/examples/fadot.c
@@ -66,7 +66,7 @@ int main (int argc, char **argv) {
 
   progname = argv[0];
 
-  while ((c = getopt (argc, argv, "nf:o:")) != -1)
+  while ((c = getopt (argc, argv, "nhf:o:")) != -1)
     switch (c)
       {
       case 'n':
@@ -75,6 +75,9 @@ int main (int argc, char **argv) {
       case 'f':
         file_output = optarg;
         break;
+      case 'h':
+        usage();
+        break;
       case 'o':
         operation = optarg;
         break;
-- 
1.7.7.6

